protos/atomic_swap.proto

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 REMME
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+syntax = "proto3";
+
+message AtomicSwapMethod {
+    enum Method {
+        INIT = 0;
+        APPROVE = 1;
+        EXPIRE = 2;
+        SET_SECRET_LOCK = 3;
+        CLOSE = 4;
+    }
+}
+
+message AtomicSwapInitPayload {
+    string receiver_address = 1;
+    string sender_address_non_local = 7;
+    uint64 amount = 2;
+    string swap_id = 3;
+    string secret_lock_optional_bob = 4;
+    string email_address_encrypted_optional_alice = 5;
+    uint32 created_at = 6;
+}
+
+message AtomicSwapApprovePayload {
```

Comment:
Do we really need two payloads with same contents?

Reply:
Would you like to give an example and how it will ease the development process and readability?